src/PowerShellEditorServices/Services/Analysis/AnalysisService.cs

```diff
@@ -725,158 +207,129 @@ private class PowerShellResult
-            SemaphoreSlim fileLock;
-            Dictionary<string, MarkerCorrection> fileCorrections;
-            bool newEntryNeeded = false;
-            if (_mostRecentCorrectionsByFile.TryGetValue(scriptFile.DocumentUri, out (SemaphoreSlim, Dictionary<string, MarkerCorrection>) fileCorrectionsEntry))
-            {
-                fileLock = fileCorrectionsEntry.Item1;
-                fileCorrections = fileCorrectionsEntry.Item2;
-            }
-            else
-            {
-                fileLock = new SemaphoreSlim(initialCount: 1, maxCount: 1);
-                fileCorrections = new Dictionary<string, MarkerCorrection>();
-                newEntryNeeded = true;
-            }
-
-            fileLock.Wait();
-            try
-            {
-                if (newEntryNeeded)
-                {
-                    // If we create a new entry, we should do it after acquiring the lock we just created
-                    // to ensure a competing thread can never acquire it first and read invalid information from it
-                    _mostRecentCorrectionsByFile[scriptFile.DocumentUri] = (fileLock, fileCorrections);
-                }
-                else
-                {
-                    // Otherwise we need to clear the stale corrections
-                    fileCorrections.Clear();
-                }
+                    if (!ct.CanBeCanceled || ct.IsCancellationRequested)
+                    {
+                        break;
+                    }
 
-                foreach (ScriptFileMarker marker in markers)
-                {
-                    // Does the marker contain a correction?
-                    Diagnostic markerDiagnostic = GetDiagnosticFromMarker(marker);
-                    if (marker.Correction != null)
+                    // Create the entry for this file if it does not already exist
+                    SemaphoreSlim fileLock;
+                    Dictionary<string, MarkerCorrection> fileCorrections;
+                    bool newEntryNeeded = false;
+                    if (_mostRecentCorrectionsByFile.TryGetValue(file.DocumentUri, out (SemaphoreSlim, Dictionary<string, MarkerCorrection>) fileCorrectionsEntry))
                     {
-                        string diagnosticId = GetUniqueIdFromDiagnostic(markerDiagnostic);
-                        fileCorrections[diagnosticId] = marker.Correction;
+                        fileLock = fileCorrectionsEntry.Item1;
+                        fileCorrections = fileCorrectionsEntry.Item2;
+                    }
+                    else
+                    {
+                        fileLock = AsyncUtils.CreateSimpleLockingSemaphore();
+                        fileCorrections = new Dictionary<string, MarkerCorrection>();
+                        newEntryNeeded = true;
                     }
 
-                    diagnostics.Add(markerDiagnostic);
+                    await fileLock.WaitAsync();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/app/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4416818)